Source/santad/SNTPolicyProcessor.mm

```diff
@@ -44,6 +45,123 @@ - (instancetype)initWithRuleTable:(SNTRuleTable *)ruleTable {
+      {{SNTRuleTypeCDHash, SNTRuleStateAllowTransitive}, SNTEventStateAllowTransitive},
+      {{SNTRuleTypeCDHash, SNTRuleStateBlock}, SNTEventStateBlockCDHash},
+      {{SNTRuleTypeBinary, SNTRuleStateAllow}, SNTEventStateAllowBinary},
+      {{SNTRuleTypeBinary, SNTRuleStateAllowTransitive}, SNTEventStateAllowTransitive},
+      {{SNTRuleTypeBinary, SNTRuleStateAllowCompiler}, SNTEventStateAllowCompiler},
+      {{SNTRuleTypeBinary, SNTRuleStateSilentBlock}, SNTEventStateBlockBinary},
+      {{SNTRuleTypeBinary, SNTRuleStateBlock}, SNTEventStateBlockBinary},
+      {{SNTRuleTypeSigningID, SNTRuleStateAllow}, SNTEventStateAllowSigningID},
+      {{SNTRuleTypeSigningID, SNTRuleStateAllowCompiler}, SNTEventStateAllowCompiler},
+      {{SNTRuleTypeSigningID, SNTRuleStateSilentBlock}, SNTEventStateBlockSigningID},
+      {{SNTRuleTypeSigningID, SNTRuleStateBlock}, SNTEventStateBlockSigningID},
+      {{SNTRuleTypeCertificate, SNTRuleStateAllow}, SNTEventStateAllowCertificate},
+      {{SNTRuleTypeCertificate, SNTRuleStateSilentBlock}, SNTEventStateBlockCertificate},
+      {{SNTRuleTypeCertificate, SNTRuleStateBlock}, SNTEventStateBlockCertificate},
+      {{SNTRuleTypeTeamID, SNTRuleStateAllow}, SNTEventStateAllowTeamID},
+      {{SNTRuleTypeTeamID, SNTRuleStateSilentBlock}, SNTEventStateBlockTeamID},
+      {{SNTRuleTypeTeamID, SNTRuleStateBlock}, SNTEventStateBlockTeamID},
+    };
+
+  auto iterator = decisions.find(std::pair<SNTRuleType, SNTRuleState>{rule.type, rule.state});
+  if (iterator != decisions.end()) {
+    cd.decision = iterator->second;
+  } else {
+    // We don't have a matching rule type, rule state for this.
+    if (mode == SNTClientModeMonitor) {
+      cd.decision = SNTEventStateAllowUnknown;
+    } else {
+      cd.decision = SNTEventStateBlockUnknown;
+    }
+  }
+
+  if (rule.state == SNTRuleStateSilentBlock) {
+    cd.silentBlock = YES;
+  } else if (rule.state == SNTRuleStateAllowCompiler) {
+    if (!enableTransitiveRules) {
+      switch (rule.type) {
+        case SNTRuleTypeCDHash: cd.decision = SNTEventStateAllowCDHash; break;
+        case SNTRuleTypeBinary: cd.decision = SNTEventStateAllowBinary; break;
+        case SNTRuleTypeSigningID: cd.decision = SNTEventStateAllowSigningID; break;
+        default:
+          // Programming error. Something's marked as a compiler that shouldn't
+          // be.
+          LOGE(@"Invalid compiler rule type %ld", rule.type);
+          [NSException
+             raise:@"Invalid compiler rule type"
+            format:@"updateCachedDecisionForRule: Unexpected compiler rule type: %ld", rule.type];
+          break;
+      }
+    }
+  } else if (rule.state == SNTRuleStateAllowTransitive) {
+    // If transitive rules are enabled, then SNTRuleStateAllowTransitive rules
+    // become SNTEventStateAllowTransitive decisions.  Otherwise, we treat the
+    // rule as if it were SNTRuleStateUnknown.
+    if (!enableTransitiveRules) {
```

Comment:
Should this be `if (enableTransitiveRules)`?

Reply:
Nope the map takes care of this for us, but the comment should probably be updated. This is to fallback in the event that we're not enabling transitive rules but have transitive states from the database.

Reply:
Ack. The extra comment makes it clearer